pkg/tfbridge/info.go

```diff
@@ -111,6 +114,31 @@ type ProviderInfo struct {
 	SkipExamples func(SkipExamplesArgs) bool
 }
 
+// DocRuleInfo controls file discovery and edits for any subset of docs in a provider.
+type DocRuleInfo struct {
```

Comment:
There's some overlap with DocInfo at least conceptually but I think I see why you'd want this at top level.